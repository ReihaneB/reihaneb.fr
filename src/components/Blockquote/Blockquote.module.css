.root {
  padding: 16px;
  background-color: var(--grey-light);
  display: grid;
  gap: 16px;
  background-size: 8px 8px;
  border-radius: 16px;
}

.root::before{
  --quote-height: 48px;
  content: "";
  margin-left: calc(var(--padding) * -1);
  margin-top: calc(var(--padding) * -1);
  height: var(--quote-height);
  width: calc(var(--quote-height) * 1.1);
  background-image:
    radial-gradient(
      circle at bottom right,
      transparent calc(var(--quote-height) / 4 - 1px),
      var(--accent-color, black) calc(var(--quote-height) / 4) calc(var(--quote-height) / 2), 
      transparent calc(var(--quote-height) / 2 + 1px)
    ),
    linear-gradient(var(--accent-color, black), var(--accent-color, black));
  background-size: calc(var(--quote-height) / 2) calc(var(--quote-height) / 2);
  background-position: top left, bottom left;
  background-repeat: space no-repeat
}

.text {
  margin-left: 16px;
  font-style: italic;
}
