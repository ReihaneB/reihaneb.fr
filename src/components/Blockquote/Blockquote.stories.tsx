import type { Meta, StoryObj } from '@storybook/react';

import Blockquote from './Blockquote';

const meta: Meta<typeof Blockquote> = {
  title: 'Components/Blockquote',
  component: Blockquote,
};

export default meta;
type Story = StoryObj<typeof Blockquote>;

const defaultArgs = {
  children: 'No one is perfect - thatâ€™s why pencils have erasers.',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const LongQuote: Story = {
  args: {
    children: (
      `
        Et ad commodo ea ullamco quis irure consectetur sit est elit.
        Commodo amet aliqua incididunt id magna quis consequat aliqua consectetur ad.
        Laborum culpa quis elit eu ad. Est ex elit adipisicing pariatur sunt ipsum nostrud dolore.
        Lorem minim incididunt ullamco laboris magna labore enim.
        Ea qui anim fugiat proident deserunt exercitation id ad enim.
        Duis nulla ut ipsum excepteur pariatur pariatur mollit ex.
      `
    ),
  },
};

export const ShortQuote: Story = {
  args: {
    children: 'No one is perfect.',
  },
};
