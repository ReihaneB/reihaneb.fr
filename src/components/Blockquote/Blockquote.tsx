import { BlockquoteProps } from './Blockquote.d';
import styles from './Blockquote.module.css';

function Blockquote({
  children,
  ...rest
}: BlockquoteProps) {
  return (
    <blockquote
      className={styles.root}
      {...rest}
    >
      <p className={styles.text}>
        {children}
      </p>
    </blockquote>
  );
}

export default Blockquote;
