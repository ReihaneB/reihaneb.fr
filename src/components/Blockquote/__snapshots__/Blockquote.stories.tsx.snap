// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Blockquote LongQuote smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <blockquote class="src-components-Blockquote-Blockquote-module__root--LtAGd">
    <p class="src-components-Blockquote-Blockquote-module__text--H_BLP">
      Et ad commodo ea ullamco quis irure consectetur sit est elit.
        Commodo amet aliqua incididunt id magna quis consequat aliqua consectetur ad.
        Laborum culpa quis elit eu ad. Est ex elit adipisicing pariatur sunt ipsum nostrud dolore.
        Lorem minim incididunt ullamco laboris magna labore enim.
        Ea qui anim fugiat proident deserunt exercitation id ad enim.
        Duis nulla ut ipsum excepteur pariatur pariatur mollit ex.
    </p>
  </blockquote>
</main>
`;

exports[`Components/Blockquote Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <blockquote class="src-components-Blockquote-Blockquote-module__root--LtAGd">
    <p class="src-components-Blockquote-Blockquote-module__text--H_BLP">
      No one is perfect - thatâ€™s why pencils have erasers.
    </p>
  </blockquote>
</main>
`;

exports[`Components/Blockquote ShortQuote smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <blockquote class="src-components-Blockquote-Blockquote-module__root--LtAGd">
    <p class="src-components-Blockquote-Blockquote-module__text--H_BLP">
      No one is perfect.
    </p>
  </blockquote>
</main>
`;
